func_test.go

```diff
@@ -9,6 +9,102 @@ import (
 	"testing"
 )
 
+type testStruct0 struct{ field0 int }
+type testStruct1 struct{ field1 int }
+
+// Disable unused checking for testInterface, as we're just using it to make
+// sure that our func typechecking works properly, and we don't need to
+// call it to do so.
+// nolint:unused
+type testInterface interface{ FuncTestMethod() }
+type testInterfaceImpl struct{}
+
+func (s *testInterfaceImpl) FuncTestMethod() {}
+
+var fnTestNilFuncArgs = Func(
+	func(i int, s string, ss []string, m map[int]int,
+		ts0 testStruct0, pts1 *testStruct1, ti testInterface) Slice {
+
+		return Const(1, []int{})
+	})
+
+// TestNilFuncArgs verifies that Func invocation handles untyped nil arguments
+// properly.
+func TestNilFuncArgs(t *testing.T) {
+	ts0 := testStruct0{field0: 0}
+	pts1 := &testStruct1{field1: 0}
+	ptii := &testInterfaceImpl{}
+	for _, c := range []struct {
+		name string
+		args []interface{}
+		ok   bool
+	}{
+		{
+			name: "all non-nil",
+			args: []interface{}{
+				0, "", []string{}, map[int]int{0: 0},
+				ts0, pts1, ptii,
+			},
+			ok: true,
+		},
+		{
+			name: "nil for types that can be nil",
+			args: []interface{}{
+				0, "", nil, nil,
+				ts0, nil, nil,
+			},
+			ok: true,
+		},
+		{
+			name: "nil for int",
+			args: []interface{}{
+				nil, "", []string{}, map[int]int{0: 0},
+				ts0, pts1, ptii,
+			},
+			ok: false,
+		},
+		{
+			name: "nil for string",
+			args: []interface{}{
+				0, nil, []string{}, map[int]int{0: 0},
+				ts0, pts1, ptii,
+			},
+			ok: false,
+		},
+		{
+			name: "nil for struct",
+			args: []interface{}{
+				0, "", []string{}, map[int]int{0: 0},
+				nil, pts1, ptii,
+			},
+			ok: false,
+		},
+	} {
+		t.Run(c.name, func(t *testing.T) {
+			checkPanic := func() {
```

Comment:
Optional: Use [`assert.Panics`](https://pkg.go.dev/github.com/stretchr/testify/assert?tab=doc#Panics) and its kin.